File: app/javascript/channels/test1_channel.js
import consumer from "channels/consumer";

consumer.subscriptions.create("Test1Channel", {
  connected() {
    // Called when the subscription is ready for use on the server
    this.send({ message: "Client is live" });
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    console.log(data);
  },
});

File: app/channels/test1_channel.rb
class Test1Channel < ApplicationCable::Channel
  def subscribed
    # stream_from "some_channel"
    stream_from 'test1'
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end
  
  def receive(data)
    puts data["message"]
    ActionCable.server.broadcast("test", "ActionCable is connected")
  end
end
